python/pyarrow/compute.py

```diff
@@ -113,6 +113,24 @@ def func(left, right):
 multiply = _simple_binary_function('multiply')
 
 
+def contains_exact(array, pattern):
+    """
+    Check whether a pattern occurs as part of the values of the array.
```

Comment:
That's rather vague. Does `contains_exact(pa.array([1,2,3]), pa.array([1,2]))` return true? I don't think so.